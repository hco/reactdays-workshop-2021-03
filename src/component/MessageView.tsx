import React from "react";
import { Message } from "../domain/Message";

interface Props {
  message: Message;
}

export const MessageView: React.FunctionComponent<Props> = ({ message }) => {
  return (
    <article>
      {message.message}
      <footer>{message.author.toLowerCase()}</footer>
    </article>
  );
};

export default MessageView;
