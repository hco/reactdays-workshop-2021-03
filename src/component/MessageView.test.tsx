import { render } from "@testing-library/react";
import { Message } from "../domain/Message";
import MessageView from "./MessageView";

const testMessage: Message = {
  author: "Moritz",
  message: "Hallo Jest!",
  date: 132,
  id: "asdjsjasdjadsjia1",
};

describe("<MessageView />", () => {
  test("renders without crash", () => {
    render(<MessageView message={testMessage} />);
  });

  test("renders the message text on the screen", async () => {
    const { queryByText } = render(<MessageView message={testMessage} />);
    expect(queryByText("Hallo Jest!")).not.toBeNull();
  });

  test("renders the author name on the screen", async () => {
    const { queryByText } = render(<MessageView message={testMessage} />);
    expect(queryByText("moritz")).not.toBeNull();
  });
  test("renders according to snapshot", () => {
    const { container } = render(<MessageView message={testMessage} />);

    expect(container).toMatchSnapshot();
  });
});
