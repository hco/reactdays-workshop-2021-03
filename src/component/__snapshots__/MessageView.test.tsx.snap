// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageView /> renders according to snapshot 1`] = `
<div>
  <article>
    Hallo Jest!
    <footer>
      moritz
    </footer>
  </article>
</div>
`;
